src/main/java/org/opensearch/ad/AnomalyDetectorExtension.java

```diff
@@ -89,18 +87,16 @@ public List<Setting<?>> getSettings() {
     }
 
     @Override
-    public Collection<Object> createComponents(SDKClient sdkClient, ClusterService clusterService, ThreadPool threadPool) {
-        return null;
-    }
-
-    private static ExtensionSettings initializeSettings() throws IOException {
-        ExtensionSettings settings = Extension.readSettingsFromYaml(EXTENSION_SETTINGS_PATH);
-        if (settings == null || settings.getHostAddress() == null || settings.getHostPort() == null) {
-            throw new IOException("Failed to initialize Extension settings. No port bound.");
-        }
-        return settings;
+    public List<NamedXContentRegistry.Entry> getNamedXContent() {
+        // Copied from AnomalyDetectorPlugin getNamedXContent
+        return ImmutableList.of(AnomalyDetector.XCONTENT_REGISTRY, AnomalyResult.XCONTENT_REGISTRY, DetectorInternalState.XCONTENT_REGISTRY
+        // Pending Job Scheduler Integration
```

Comment:
Can we add this in https://github.com/opensearch-project/opensearch-sdk-java/issues/216?

Reply:
Sure!